liquidator/src/liquidate.rs

```diff
@@ -596,7 +604,24 @@ impl<'a> LiquidateHelper<'a> {
         // }
 
         if let Some(txsig) = self.token_liq().await? {
-            return Ok(txsig);
+            return Ok(Some(txsig));
+        }
+
+        if self.health_cache.has_perp_open_fills() {
+            log::info!(
+                "Account {} has open perp fills, maint_health {}, waiting...",
+                self.pubkey,
+                self.maint_health
+            );
+            return Ok(None);
+        }
+
+        if self.health_cache.has_phase2_liquidatable() {
```

Comment:
technically this check would cover the one on 610, but ok the one on 610 comes with an explicit msg about open perp fills

Reply:
The key difference is that 610 is info-log-and-wait, while this is log-critical-error